python_modules/dagster/dagster/_core/definitions/asset_graph.py

```diff
@@ -751,10 +757,10 @@ def __init__(
         self._asset_graph = asset_graph
         self._include_required_multi_assets = include_required_multi_assets
 
-        toposorted_asset_keys = asset_graph.toposort_asset_keys()
+        asset_keys_by_level = toposort.toposort(asset_graph.asset_dep_graph["upstream"])
```

Comment:
considering we're still using this level-based system in two places (here and in the definition of the new toposorted_asset_keys property), can we just have two properties of the AssetGraph class (one with levels one without, with the non-level one using the leveled one)?

Reply:
done